lib/puppetserver/ca/revoke_action.rb

```diff
@@ -0,0 +1,149 @@
+require 'puppetserver/ca/utils'
+
+require 'optparse'
+require 'openssl'
+require 'net/https'
+require 'json'
+
+module Puppetserver
+  module Ca
+    class RevokeAction
+      SUMMARY = 'Revoke a given certificate'
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca revoke [--help|--version]
+  puppetserver ca revoke CERTNAME
+
+Description:
+Given a valid certname, instructs the CA to revoke it over
+HTTPS using the local agent's PKI
+
+Options:
+BANNER
+
+      def self.parser(parsed = {})
+        OptionParser.new do |o|
+          o.banner = BANNER
+          o.on('--config', 'Path to puppet.conf') do |conf|
```

Comment:
should this be lifted to the main cli class? I think eventually it would then need to reference both a puppet server's conf.d and a puppet.conf??? But maybe the Cli class is a better place for it for the time being regardless?

Reply:
We don't need --help here too?